README.md
# Blinking point

Easy way to create a blinking point for your Flutter project.

## Installation

Add this to your package's pubspec.yaml file:

```yaml
dependencies:
  blinking_point: ^1.0.0+2
```

## Usage

Import the file.

```dart
import 'package:blinking_point/blinking_point.dart';
```

Calling the point: 

```dart
new BlinkingPoint(
    xCoor: 100.0, // The x coordinate of the point
    yCoor: 500.0, // The y coordinate of the point
    pointColor: Colors.red, // The color of the point
    pointSize: 10.0, // The size of the point
);
```

## Demo

<img src="demo.gif" width="250" />


CHANGELOG.md
- Added new readme with demo.gif for demonstration.

test/widget_test.dart
// This is a basic Flutter widget test.
// To perform an interaction with a widget in your test, use the WidgetTester utility that Flutter
// provides. For example, you can send tap and scroll gestures. You can also use WidgetTester to
// find child widgets in the widget tree, read text, and verify that the values of widget properties
// are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:blinking_point/main.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(new MyApp());

    // Verify that our counter starts at 0.
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Tap the '+' icon and trigger a frame.
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();

    // Verify that our counter has incremented.
    expect(find.text('0'), findsNothing);
    expect(find.text('1'), findsOneWidget);
  });
}


lib/lul.dart
import 'package:flutter/material.dart';
import 'package:blinking_point/circle.dart';
import 'package:flutter/animation.dart';

final color = [
  new PointModel(
    Colors.red,
  ),
];

class BlinkingPoint extends StatefulWidget {
  final double xCoor;
  final double yCoor;
  final Color pointColor;
  final double pointSize;

  BlinkingPoint({
    this.xCoor,
    this.yCoor,
    this.pointColor,
    this.pointSize,
  });

  @override
  _BlinkingPointState createState() => new _BlinkingPointState();
}

class _BlinkingPointState extends State<BlinkingPoint>
    with SingleTickerProviderStateMixin {
  Animation animation;
  AnimationController animationController;

  @override
  void initState() {
    super.initState();
    animationController = AnimationController(
      duration: const Duration(milliseconds: 2000),
      vsync: this,
    );
    animation = Tween(begin: 0.0, end: 500.0).animate(animationController);
    animation.addStatusListener((status) {
      if (status == AnimationStatus.completed) {
        animationController.reverse();
      } else if (status == AnimationStatus.dismissed) {
        animationController.forward();
      }
    });
  }

  @override
  Widget build(BuildContext context) {
    return LogoAnimation(
      animation: animation,
    );
  }

  @override
  void dispose() {
    animationController.dispose();
    super.dispose();
  }
}

class LogoAnimation extends AnimatedWidget {
  LogoAnimation({Key key, Animation animation})
      : super(key: key, listenable: animation);

  @override
  Widget build(BuildContext context) {
    Animation animation = listenable;
    print(animation.value);
    return new CustomPaint(
      foregroundPainter: Circle(
        xCoor: 200.0,
        yCoor: 500.0,
        color: Colors.red,
        pointSize: animation.value,
      ),
    );
  }
}

class PointModel {
  final color;
  PointModel(
    this.color,
  );
}


lib/circle.dart
import 'package:flutter/material.dart';
import 'dart:ui';

class Circle extends CustomPainter {
  final Color color;
  final double pointSize;
  final double xCoor;
  final double yCoor;
  final double blinkRadius;

  Circle({
    this.color,
    this.pointSize,
    this.xCoor,
    this.yCoor,
    this.blinkRadius,
  });

  @override
  void paint(Canvas canvas, Size size) {
    // Draw the mid point
    Paint line = new Paint();
    line.strokeCap = StrokeCap.round;
    line.color = color;
    line.strokeWidth = pointSize / 2;
    Offset center = new Offset(xCoor, yCoor / 2);
    double pointRadius = pointSize / 2;

    line.style = PaintingStyle.fill;
    canvas.drawCircle(center, pointRadius, line);

    line.style = PaintingStyle.stroke;
    canvas.drawCircle(center, blinkRadius, line);
  }

  @override
  bool shouldRepaint(CustomPainter oldDelegate) {
    return true;
  }
}


lib/load_state.dart
import 'package:flutter/material.dart';
import 'package:flutter/animation.dart';
import 'package:blinking_point/circle.dart';

class LmaoApp extends StatefulWidget {
  @override
  LmaoAppState createState() => new LmaoAppState();
}

class LmaoAppState extends State<LmaoApp> with SingleTickerProviderStateMixin {
  Animation animation;
  AnimationController animationController;

  @override
  void initState() {
    super.initState();
    animationController = AnimationController(
      duration: Duration(milliseconds: 500),
      vsync: this,
    );

    animation = Tween(begin: 0.0, end: 20.0).animate(animationController);
    animation.addStatusListener((status) {
      if (status == AnimationStatus.completed) {
        animationController.reverse();
      } else if (status == AnimationStatus.dismissed) {
        animationController.forward();
      }
    });
    animationController.forward();
  }

  @override
  Widget build(BuildContext context) {
    return LogoAnimation(
      animation: animation,
    );
  }

  @override
  void dispose() {
    animationController.dispose();
    super.dispose();
  }
}

class LogoAnimation extends AnimatedWidget {
  LogoAnimation({Key key, Animation animation})
      : super(key: key, listenable: animation);

  @override
  Widget build(BuildContext context) {
    Animation animation = listenable;
    return new CustomPaint(
      foregroundPainter: Circle(
        xCoor: 200.0,
        yCoor: 500.0,
        color: Colors.red,
        pointSize: animation.value,
      ),
    );
    // return Center(
    //   child: Container(
    //     height: animation.value,
    //     width: animation.value,
    //     child: FlutterLogo(),
    //   ),
    // );
  }
}


lib/blink_point.dart
import 'package:flutter/material.dart';
import 'package:blinking_point/circle.dart';
import 'package:flutter/animation.dart';

final color = [
  new PointModel(
    Colors.red,
  ),
];

class BlinkingPoint extends StatefulWidget {
  final double xCoor;
  final double yCoor;
  final Color pointColor;
  final double pointSize;

  BlinkingPoint({
    this.xCoor,
    this.yCoor,
    this.pointColor,
    this.pointSize,
  });

  @override
  _BlinkingPointState createState() => new _BlinkingPointState();
}

class _BlinkingPointState extends State<BlinkingPoint>
    with SingleTickerProviderStateMixin {
  Animation<double> animation;
  AnimationController animationController;

  @override
  void initState() {
    super.initState();
    animationController = AnimationController(
      duration: const Duration(milliseconds: 2000),
      vsync: this,
    );
    animation = Tween(begin: 0.0, end: 500.0).animate(animationController);
    animation.addStatusListener((status) {
      if (status == AnimationStatus.completed) {
        animationController.reverse();
      } else if (status == AnimationStatus.dismissed) {
        animationController.forward();
      }
    });
  }

  @override
  void dispose() {
    animationController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return new Container(
      height: animation.value,
      width: animation.value,
      child:
          /*new CustomPaint(
          foregroundPainter: Circle(
            xCoor: widget.xCoor,
            yCoor: widget.yCoor,
            color: widget.pointColor,
            pointSize: widget.pointSize,
          ),
        ),*/
          FlutterLogo(),
    );
  }
}

class PointModel {
  final color;
  PointModel(
    this.color,
  );
}


lib/blinking_point.dart
import 'package:flutter/material.dart';
import 'package:flutter/animation.dart';
import 'package:blinking_point/circle.dart';

class BlinkingPoint extends StatefulWidget {
  final double xCoor;
  final double yCoor;
  final Color pointColor;
  final double pointSize;

  BlinkingPoint({
    this.xCoor,
    this.yCoor,
    this.pointColor,
    this.pointSize,
  });

  @override
  BlinkingPointState createState() => new BlinkingPointState();
}

class BlinkingPointState extends State<BlinkingPoint>
    with SingleTickerProviderStateMixin {
  Animation animation;
  AnimationController animationController;

  @override
  void initState() {
    super.initState();
    animationController = AnimationController(
      duration: Duration(milliseconds: 1000),
      vsync: this,
    );

    animation = Tween(begin: 0.0, end: widget.pointSize * 4)
        .animate(animationController);
    animation.addStatusListener((status) {
      if (status == AnimationStatus.completed) {
        animationController.reset();
      } else if (status == AnimationStatus.dismissed) {
        animationController.forward();
      }
    });
    animationController.forward();
  }

  @override
  Widget build(BuildContext context) {
    return LogoAnimation(
      xCoor: widget.xCoor,
      yCoor: widget.yCoor,
      pointColor: widget.pointColor,
      pointSize: widget.pointSize,
      animation: animation,
    );
  }

  @override
  void dispose() {
    animationController.dispose();
    super.dispose();
  }
}

class LogoAnimation extends AnimatedWidget {
  final double xCoor;
  final double yCoor;
  final Color pointColor;
  final double pointSize;

  LogoAnimation({
    Key key,
    Animation animation,
    this.xCoor,
    this.yCoor,
    this.pointColor,
    this.pointSize,
  }) : super(key: key, listenable: animation);

  @override
  Widget build(BuildContext context) {
    Animation animation = listenable;
    return new CustomPaint(
      foregroundPainter: Circle(
        xCoor: xCoor,
        yCoor: yCoor,
        color: pointColor,
        pointSize: pointSize,
        blinkRadius: animation.value,
      ),
    );
  }
}


lib/main.dart
import 'package:flutter/material.dart';
import 'package:blinking_point/blinking_point.dart';

void main() => runApp(new MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      title: 'Blinking Point',
      theme: new ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: new MyHomePage(title: 'BlinkingPoint'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => new _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return new Stack(
      children: [
        new BlinkingPoint(
          xCoor: 200.0,
          yCoor: 350.0,
          pointColor: Colors.red,
          pointSize: 10.0,
        ),
        new BlinkingPoint(
          xCoor: 200.0,
          yCoor: 700.0,
          pointColor: Colors.yellow,
          pointSize: 10.0,
        ),
        new BlinkingPoint(
          xCoor: 200.0,
          yCoor: 1050.0,
          pointColor: Colors.blue,
          pointSize: 10.0,
        ),
      ],
    );

    // return new LmaoApp(
    //   xCoor: 100.0,
    //   yCoor: 600.0,
    //   pointColor: Colors.red,
    //   pointSize: 20.0,
    // );
  }
}
